### DISCUSSION TITLE
Ver 1.10 agent and jaeger exporter

### DISCUSSION THREAD

**Discussion by prubinVI** [open-telemetry/opentelemetry-java#4106] [Q&A]
I am trying to deploy a spring boot java server to k8s.  we have a Jaeger collector running, I have tried many configurations and while I got the url to the Jaeger server correct, it keeps trying to use the oltp exporter:

I am using on the java command line
-javaagent:/app/libs/opentelemetry-javaagent-1.10.0.jar

I have tried environment variables, and injecting parameters into the jvm command line and neither seems to work:

  OTEL_EXPORTER: "jaeger"
  OTEL_TRACES_EXPORTER: "jaeger"

-dotel.exporter=jeager
-dotel.traces.exporter=jaeger

I verified that all of these are in the ENV and or system properties, but The open telemetry agent continues to give the error above.

`[otel.javaagent 2022-01-21 16:41:55:130 +0000] [OkHttp http://jaeger-collector.monitoring-system:14268/...] ERROR io.opentelemetry.exporter.otlp.internal.grpc.OkHttpGrpcExporter - Failed to export spans. The request could not be executed. Full error message: null
`

The full error message null is not very helpful and the exporter being the oltp exporter is just confusing.

Can you suggest anything to help me diagnose what I am doing wrong?

Thanks
Paul.

**Comment by trask** [comment]
hi @prubinVI! `io.opentelemetry.exporter.otlp.internal.grpc.OkHttpGrpcExporter` class is used by all of the grpc-based exporters (including jaeger). I have opened #4107 to move this class out of the "otlp" package to avoid this confusion.

**Reply by prubinVI** [reply]
Thanks any way I can find out with the "null" error is?

**Reply by anuraaga** [reply]
Hi @prubinVI - that log message is outputed when no connection could be made to the endpoint at all I believe.

https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/grpc/OkHttpGrpcExporter.java#L106

It looks like we could but aren't including the exception's stack trace in that log, will fix that, but in the meantime it looks like the exception message isn't set by the HTTP library and we end up outputing null. I would confirm that you have connectivity to that port and that HTTP is correct instead of HTTPS for example.

**Reply by prubinVI** [reply]
Thank you, I am investigating.

**Reply by prubinVI** [reply]
ok, I got into the pod, the url appears good:

`/usr/bin/curl
/ # nc jaeger-collector.monitoring-system 14268

HTTP/1.1 400 Bad Request
Content-Type: text/plain; charset=utf-8
Connection: close

400 Bad Request`

Suggestions?

**Reply by prubinVI** [reply]
More info I set OTEL_PROPAGATORS to b3 and the error change to:

`[otel.javaagent 2022-01-24 13:47:13:138 +0000] [OkHttp http://jaeger-collector.monitoring-system:14268/...] ERROR io.opentelemetry.exporter.otlp.internal.grpc.OkHttpGrpcExporter - Failed to export spans. The request could not be executed. Full error message: exhausted all routes`

**Reply by prubinVI** [reply]
turns out that was the "thrift' port, they switched me to the 'grcp' port and all is good now, thank you for your assistance

**Comment by GarikVologodskiy** [comment]
Hello!
javaagent version 1.12.1 the same issue (everything is OK with the agent version 0.7.0). Could you help me with it please?
Export to jaeger

**error**
io.opentelemetry.exporter.internal.grpc.OkHttpGrpcExporter - Failed to export spans. The request could not be executed. Full error message: Failed to connect to localhost/127.0.0.1:4317

**start**
JAVA_OPTS="${JAVA_OPTS} \
  -Xms${JAVA_XMS} \
  -Xmx${JAVA_XMX} \
  -Dapplication.name=${APP_NAME} \
  -Dapplication.home=${APP_HOME} \
  -Dotel.exporter=jaeger \
  -Dotel.jaeger.endpoint=jaeger-all-in-one:14250 \
  -Dotel.jaeger.service.name=spring-boot-rest-service \
  -javaagent:${APP_HOME}/opentelemetry-javaagent.jar"

exec java ${JAVA_OPTS} \
  -jar "${APP_HOME}/${APP_NAME}.jar" \
  --spring.config.location=/config/application.yml

**Jaeger**
  jaeger-all-in-one:
    image: "jaegertracing/all-in-one:latest"
    ports:
      - 5775:5775/udp
      - 6831:6831/udp
      - 6832:6832/udp
      - 5778:5778
      - 16686:16686
      - 14268:14268
      - 14250:14250

**Reply by trask** [reply]
merging thread into #https://github.com/open-telemetry/opentelemetry-java/issues/4279#issuecomment-1073063875
